### PR TITLE
[M68k] allow 16-bit registers for MOVE to/from CCR

### LINKED ISSUE #106210: [M68k] incorrect assertion on move from CC register to data register
The assertion "Need DR8 register to copy CCR" in M68kInstrInfo::copyPhysReg appears to be wrong. According to the Programmer's Reference Manual, MOVE from CCR takes a 32bit destination operand and zero extends the 5 bits in the CC register.

### PR DESCRIPTION
**Pull request by TechnoElf**
Fixes #106210.

### CODE REVIEW

**Comment by TechnoElf** [comment]
Ping @0x59616e

File: llvm/lib/Target/M68k/M68kInstrInfo.cpp [hunk 1]
```diff
@@ -709,13 +709,27 @@ void M68kInstrInfo::copyPhysReg(MachineBasicBlock &MBB,
   bool ToSR = DstReg == M68k::SR;
 
   if (FromCCR) {
-    assert(M68k::DR8RegClass.contains(DstReg) &&
-           "Need DR8 register to copy CCR");
-    Opc = M68k::MOV8dc;
+    if (M68k::DR8RegClass.contains(DstReg))
```
Comment (mshockwave) [inline llvm/lib/Target/M68k/M68kInstrInfo.cpp:712]: please add braces when one of the else/else if arms is using braces: https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements

**Review by mshockwave — COMMENTED** [review]
Thanks for the patch! Could you add some tests?

**Comment by TechnoElf** [comment]
Should I close this PR as #114714 already implements the requested changes?

**Comment by mshockwave** [comment]
> Should I close this PR as #114714 already implements the requested changes?

Feel free to do that!

**Pull request closed without merging**